### ISSUE TITLE
Support for reading data from Loom files

### ISSUE THREAD

**Issue by pcm32** [maximilianh/cellBrowser#46]
It would be great if in the near future cellBrowser could read loom files (https://linnarssonlab.org/loompy/format/index.html) as input, either expecting t-SNE coordinates in a particular slot of loom or allowing the user to choose the slot where the coordinates are.

We are working towards interoperability of tools and this would great simplify things!

**Comment by maximilianh** [comment]
Hi Pablo, interesting idea! Is there a converter that comes with loom that
does a loom -> .tsv.gz conversion? If not, would it be sufficient to
convert to that format? Otherwise, is there a sample piece of code that
shows how to get the expression vector for a gene ? I looked in their
cookbook, but can't find anything, here:
https://linnarssonlab.org/loompy/cookbook/index.html

**Comment by pcm32** [comment]
The central object in loom is the expression matrix (genes vs samples/cells), I presume that scan and map methods, part of LoomConnect, should give you access to a gene by gene slicing of the matrix, producing an expression vector for each gene.

Outputting loom I'm sure it would be very useful for some use cases, although I was more thinking on visualising the output of clustering tools that are produced in loom, so more of a Loom reading feature.

**Comment by maximilianh** [comment]
Yes, I understand that we're mostly interested in reading loom files, not
writing them.  So there is no loom -> text converter available anywhere
that you know of?

I've never seen a loom file in the wild. Do you have an example?

Given that it's a format for genes and cells, it's surprising that the loom
documentation does not give an example on how to get a the cell-vector for
a gene. The documentation talks about indices, in x and y, but not genes or
cells. I can't even figure out if this is a gene x cell or cell x gene
matrix. I am happy to add the option of reading other formats to the
cellbrowser, but it would be nice if you could provide a few lines of code
that can yield the expression vectors for all genes...
